pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -115,6 +117,69 @@ def test_groupby_aggregation_mixed_dtype():
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_aggregation_non_numeric_dtype():
+    # GH #43108
+    df = DataFrame(
+        [["M", [1]], ["M", [1]], ["W", [10]], ["W", [20]]], columns=["MW", "v"]
+    )
+
+    expected = DataFrame(
+        {
+            "v": [[1, 1], [10, 20]],
+        },
+        index=Index(["M", "W"], dtype="object", name="MW"),
+    )
+
+    with tm.assert_produces_warning(UserWarning):
+        g = df.groupby(by=["MW"])
```

Comment:
nitpick: can you call this `gb` instead of `g`?  1-letter variable names make things harder to grep for